plugins/modules/content_template.py

```diff
@@ -0,0 +1,236 @@
+          O(resource_pool) must belong to O(cluster).
+        - If O(cluster) and O(host) are both specified, O(host) must be a member of O(cluster).
+        - This attribute was added in vSphere API 6.8.
+        type: str
+    folder:
+        description:
+        - Virtual machine folder into which the virtual machine template should be placed.
+        - This attribute was added in vSphere API 6.8.
+        - If not specified, the virtual machine template will be placed in the same
+          folder as the source virtual machine.
+        type: str
+    state:
+        description:
+        - State of the template in content library.
+        - If C(present), the template will be created in content library.
+        - If C(absent), the template will be deleted from content library.
+        type: str
+        default: present
+        choices:
+        - present
+        - absent
+extends_documentation_fragment:
+- vmware.vmware.vmware_rest_client.documentation
+'''
+
+EXAMPLES = r'''
+- name: Create template in content library from Virtual Machine
+  vmware.vmware.content_template:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    template: mytemplate
+    library: mylibrary
+    vm_name: myvm
+    host: myhost
+'''
+
+RETURN = r'''
+template_info:
+  description: Template creation message and template_id
+  returned: on success
+  type: dict
+  sample: {
+        "msg": "Template 'mytemplate'.",
+        "template_id": "template-1009"
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.vmware.vmware.plugins.module_utils.vmware_rest_client import VmwareRestClient
+from ansible.module_utils._text import to_native
+
+HAS_VAUTOMATION_PYTHON_SDK = False
+try:
+    from com.vmware.vcenter.vm_template_client import LibraryItems
+    from com.vmware.vapi.std.errors_client import Error
+    HAS_VAUTOMATION_PYTHON_SDK = True
+except ImportError:
+    pass
+
+
+class VmwareContentTemplate(VmwareRestClient):
+    def __init__(self, module):
+        """Constructor."""
+        super(VmwareContentTemplate, self).__init__(module)
+
+        # Initialize member variables
+        self.module = module
+        self._template_service = self.api_client.vcenter.vm_template.LibraryItems
+        self.result = {'changed': False}
+
+        # Get parameters
+        self.template = self.params.get('template')
+        self.host = self.params.get('host')
+        self.cluster = self.params.get('cluster')
+        self.resource_pool = self.params.get('resource_pool')
+        self.library = self.params.get('library')
+        self.folder = self.params.get('folder')
+        self.vm_name = self.params.get('vm_name')
+
+    def create_template_from_vm(self):
+        template = self.get_library_item_from_content_library_name(self.template, self.library)
+        if template:
+            self.result['template_info'] = dict(
```

Comment:
This will be an error? I think we should throw an error if it's update case

Reply:
This shouldn't be error. This should return `changed=False` and info that the template already exists.

Reply:
if this is the same object I agree with you, but what about editing some fields?

Reply:
What fields do you mean? Name can be updated, and in future, would be nice to support it.

Reply:
I mean if user will apply changes in name or library it will not change nothing but will response that everything is good

Reply:
If the user changes something in the placement spec but keeps the name that same, I think this module will give them a confusing result. It will return OK when in reality the template does not have the configuration they expect

Reply:
As far as I understand the placement is used only for the import process. So after import is done, it really doesn't make sense to change the field. So IMHO it's better to succed, as nothing was really changed then to fail, but makes sense to document it well, so users understand.